code/front/js/cart.js

```diff
@@ -205,8 +228,13 @@ function updateQuantityCartItem () {
             inputListener.getAttribute('value');
             //cart loop
             for (let i = 0; i < cart.length; i++) {
-                itemQuantity.value = event.target.value;
-                cart[i].quantity = itemQuantity[i].value;
+                if (cart.length === 1) {
```

Comment:
here you have to use the same approach as for deletecart item

imagine you update the "fat milk 2l" item on the cart and on the cart you have as well "fat milk 1l". You need to know which item you want to update the quantity. You need to get the respective item by id and color from the cart and update its quantity.
```
let updatedItem = event.target;
 let color = updatedItem.closest('.cart__item').getAttribute('data-color');
let productId = updatedItem.closest('.cart__item').getAttribute('data-id');
```

Reply:
Hey Faizal, Just to clarify - if I add one product ID with two different colour options -  (a) white (b) black - the product Total in cart = 2. The products are stored in the cart array as two separate objects(products) like so - [  {productID: "1" colour: "white"}, { productID: "2" color: "black" } ] 

Is it necessary to target the data-ID attribute, and the data-color attribute ? I can successfully update the products in the cart without doing so as it stands - and all tests pass.

Will this refactoring suggestion will allow me **not** to have to wrap the logic for updateQuantityCartItem in if else? I don't understand.

Reply:
will mark this as unnecessary as we discussed in the workplace